drivers/input/touchscreen/synaptics_driver_s3320.c

```diff
@@ -3952,8 +3952,8 @@ static int synaptics_ts_probe(struct i2c_client *client, const struct i2c_device
 	TP_FW = CURRENT_FIRMWARE_ID;
 	sprintf(ts->fw_id,"0x%x",TP_FW);
 
-	memset(ts->fw_name,TP_FW_NAME_MAX_LEN,0);
-	memset(ts->test_limit_name,TP_FW_NAME_MAX_LEN,0);
+	memset(ts->fw_name,0,TP_FW_NAME_MAX_LEN);
```

Comment:
This isn't a build-break issue, it's a totally separate one.  Don't lump multiple different things into the same patch, it makes it hard to apply, and the upstream kernel developers will flat out reject it.